src/gui/maptools/qgsbeziermarker.cpp

```diff
@@ -0,0 +1,291 @@
+  marker->setPenWidth( 2 );
+  return marker;
+}
+
+QgsVertexMarker *QgsBezierMarker::createHandleMarker()
+{
+  QgsVertexMarker *marker = new QgsVertexMarker( mCanvas );
+  marker->setIconType( QgsVertexMarker::ICON_CIRCLE );
+  marker->setIconSize( 8 );
+  QColor lineColor = QgsSettingsRegistryCore::settingsDigitizingLineColor->value();
+  int h, s, v, a;
+  lineColor.getHsv( &h, &s, &v, &a );
+  QColor handleColor = QColor::fromHsv( ( h + 120 ) % 360, s, v, a );
+  marker->setColor( handleColor );
+  QColor fillColor = handleColor;
+  fillColor.setAlpha( 100 );
+  marker->setFillColor( fillColor );
+  marker->setPenWidth( 2 );
+  return marker;
+}
+
+QgsRubberBand *QgsBezierMarker::createHandleLine()
+{
+  QgsRubberBand *rb = new QgsRubberBand( mCanvas, Qgis::GeometryType::Line );
+  QColor lineColor = QgsSettingsRegistryCore::settingsDigitizingLineColor->value();
+  lineColor.setAlpha( 150 );
+  rb->setColor( lineColor );
+  rb->setWidth( 1 );
+  rb->setLineStyle( Qt::DashLine );
+  return rb;
+}
+
+void QgsBezierMarker::updateFromData( const QgsBezierData &data )
+{
+  updateAnchorMarkers( data );
+  updateHandleMarkers( data );
+  updateHandleLines( data );
+  updateCurve( data );
+}
+
+void QgsBezierMarker::updateCurve( const QgsBezierData &data )
+{
+  mCurveRubberBand->reset( Qgis::GeometryType::Line );
+
+  if ( data.anchorCount() < 1 )
+    return;
+
+  QgsPointSequence points = data.interpolate();
+
+  for ( const QgsPoint &pt : std::as_const( points ) )
+  {
+    mCurveRubberBand->addPoint( QgsPointXY( pt ) );
+  }
+
+  mCurveRubberBand->setVisible( mVisible );
+}
+
+void QgsBezierMarker::updateAnchorMarkers( const QgsBezierData &data )
+{
+  while ( mAnchorMarkers.count() < data.anchorCount() )
+  {
+    mAnchorMarkers.append( createAnchorMarker() );
+  }
+  while ( mAnchorMarkers.count() > data.anchorCount() )
+  {
+    delete mAnchorMarkers.takeLast();
+  }
+
+  const QColor snapColor = QgsSettingsRegistryCore::settingsDigitizingSnapColor->value();
+  QColor snapFillColor = snapColor;
+  snapFillColor.setAlpha( 100 );
+
+  for ( int i = 0; i < data.anchorCount(); ++i )
+  {
+    const QgsPoint &anchor = data.anchor( i );
+    mAnchorMarkers[i]->setCenter( QgsPointXY( anchor ) );
+    mAnchorMarkers[i]->setVisible( mVisible );
+
+    if ( i == mHighlightedAnchor )
+    {
+      const QColor selColor = mCanvas->selectionColor();
+      mAnchorMarkers[i]->setColor( selColor );
+      QColor selFillColor = selColor;
+      selFillColor.setAlpha( 150 );
+      mAnchorMarkers[i]->setFillColor( selFillColor );
+    }
+    else
+    {
+      mAnchorMarkers[i]->setColor( snapColor );
+      mAnchorMarkers[i]->setFillColor( snapFillColor );
+    }
+  }
+}
+
+void QgsBezierMarker::updateHandleMarkers( const QgsBezierData &data )
+{
+  while ( mHandleMarkers.count() < data.handleCount() )
+  {
+    mHandleMarkers.append( createHandleMarker() );
+  }
+  while ( mHandleMarkers.count() > data.handleCount() )
```

Comment:
make mHandleMarkers a QVector<std::unique_ptr<QgsVertexMarker>> to save manual deallocation ?